spec/lib/citation_string_processor_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+require 'rails_helper'
+require './lib/emory/citation_string_processor'
+
+RSpec.describe CitationStringProcessor, type: :helper do
+  describe '#append_string_with_comma' do
+    it 'appends a comma if field present' do
+      expect(helper.append_string_with_comma(["blah"])).to eq("blah, ")
+    end
+
+    it 'returns nil if field nil' do
+      expect(helper.append_string_with_comma([""])).to be_nil
+    end
+  end
+
+  describe '#append_string_with_period' do
+    it 'appends a period if field present' do
+      expect(helper.append_string_with_period(["blah"])).to eq("blah. ")
+    end
+
+    it 'returns nil if field nil' do
+      expect(helper.append_string_with_period([""])).to be_nil
+    end
+  end
+
+  let!(:obj) { CURATE_GENERIC_WORK }
+
+  describe '#apa_edition' do
+    it 'prepends a comma and space if edition_tesim present' do
+      expect(helper.apa_edition(obj)).to eq(', 3rd ed.')
+    end
+
+    it 'returns nil if obj nil' do
```

Comment:
Can we test this method with an object that has fields besides `edition_tesim` populated as well?

Reply:
Will take care of this in next push.